pennylane/qnodes/base.py

```diff
@@ -283,11 +283,17 @@ def set_trainable_args(self, arg_indices):
                 "The trainability of arguments on immutable QNodes cannot be modified after the first evaluation."
             )
 
-        if not arg_indices:
-            # the provided arg_indices are an empty set
+        if arg_indices is None:
+            # all arguments are differentiable
             self._trainable_args = None
             return
 
+        if not arg_indices:
+            # The provided arg_indices are an empty serquence;
```

Comment:
```suggestion
            # The provided arg_indices are an empty sequence;
```
aren't they an empty *set*?